src/app/fdctl/caps.c

```diff
@@ -137,6 +154,6 @@ fd_caps_check_resource( fd_caps_ctx_t * ctx,
     rlim.rlim_cur = limit;
     rlim.rlim_max = limit;
     if( FD_UNLIKELY( setrlimit( resource, &rlim ) ) )
-      FD_LOG_ERR(( "setrlimit failed (%i-%s)", errno, fd_io_strerror( errno ) ));
+      FD_LOG_ERR(( "setrlimit failed (%i-%s) for resource %s", errno, fd_io_strerror( errno ), fd_caps_resource_str( resource ) ));
```

Comment:
I'm curious how you ended up hitting this? If your process has CAP_SYS_RESOURCE, I don't think you should hit EPERM here?

Reply:
EPERM can arise from attempting to raise `RLIMIT_NOFILE` beyond `fs.nr_open`. This isn't well known and is probably worth documenting somewhere else for the next user that hits this